### PR TITLE
network: ipv4adc: Assign IPv4 ACD address

### PR DESCRIPTION
**Pull request by holmanb**
Commit 76a86ffdbee2dd added function `ipv4acd_update_mac()` but instead invoked `ipv4ll_update_mac()`, which doesn't seem to align with debug or commit messages.

Since `ipv4acd_update_mac()` isn't called anywhere, and the failure log message under the new call to `ipv4ll_update_mac()` in 76a86ffdbee2dd states that updating the mac address in the ADC client failed, it looks like the duplicated `ipv4ll_update_mac()` was an error.

Calling `ipv4ll_update_mac()` twice consecutively doesn't appear to have been the intention of that commit.

### CODE REVIEW

**Comment by holmanb** [comment]
@yuwata Since you introduced https://github.com/systemd/systemd/commit/76a86ffdbee2dd9ef0f2b5338e14eb6ba7671456, could you please review this when you get a chance?

Also, the CI failures don't appear related to this change, and I see at least one PR with the same failure. Am I correct to assume that I can ignore them?

File: src/network/networkd-link.c [hunk 1]
```diff
@@ -2138,7 +2138,7 @@ static int link_update_hardware_address(Link *link, sd_netlink_message *message)
                         log_link_debug_errno(link, r, "Failed to manage link by its new hardware address, ignoring: %m");
         }
 
-        r = ipv4ll_update_mac(link);
+        r = ipv4acd_update_mac(link);
```
Comment (yuwata) [inline src/network/networkd-link.c:2141]: Ouch!

**Review by yuwata — APPROVED** [review]
Ouch! LGTM. Thank you.

Next time, please set a correct commit title. In this case, I will fix on merge.

**Comment by holmanb** [comment]
> Next time, please set a correct commit title. In this case, I will fix on merge.

Noted, will do. Thanks!

**Pull request merged**